internal/services/machinelearning/machine_learning_compute_instance_resource.go

```diff
@@ -215,10 +214,16 @@ func resourceComputeInstanceCreate(d *pluginsdk.ResourceData, meta interface{})
 		computeInstance.Properties.ComputeInstanceAuthorizationType = pointer.To(machinelearningcomputes.ComputeInstanceAuthorizationType(authType))
 	}
 
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.Workspaces
+	workspace, err := mlWorkspacesClient.Get(ctx, *workspaceID)
+	if err != nil {
+		return err
+	}
+
```

Comment:
```suggestion
	workspace, err := mlWorkspacesClient.Get(ctx, *workspaceID)
	if err != nil {
		return err
	}

	if workspace.Model == nil {
		return fmt.Errorf("reading %s: model was nil", *id)
	}
```

Reply:
I don't see one in the other file. I copied this logic from there. Do you think its needed?

Reply:
Yeah, I think so, if we try to pull a value from a nil return it will cause the provider to panic and crash. I have added that logic in my push to your fork.